src/main/scala/is/hail/methods/Aggregators.scala

```diff
@@ -495,3 +492,47 @@ class TakeByAggregator[T](var t: Type, var f: (Any) => Any, var n: Int)(implicit
 
   def copy() = new TakeByAggregator(t, f, n)
 }
+
+class KeyedAggregator[T, K](aggregator: TypedAggregator[T]) extends TypedAggregator[Map[Any, T]] {
+
+  private val m = mutable.Map[Any, TypedAggregator[T]]()
+
+  def result = m.map { case (k, v) => (k, v.result) }.toMap
+
+  def seqOp(x: Any) {
+    val cx = x.asInstanceOf[Row]
+    if (cx != null)
+      seqOp(cx.get(0), cx.get(1))
+    else
+      seqOp(null, null)
+  }
+
+  private def seqOp(key: Any, x: Any) {
```

Comment:
Is this ever called? It seems like all the stuff in `KeyedCodeAggregator` implements the same logic.

Reply:
It's used by the interpreter, which is the only user of the regular typed aggregator.